modules/ingester/instance_search.go

```diff
@@ -161,25 +165,38 @@ func (i *instance) searchWAL(ctx context.Context, p search.Pipeline, sr *search.
 }
 
 // searchLocalBlocks starts a search task for every local block. Must be called under lock.
-func (i *instance) searchLocalBlocks(ctx context.Context, p search.Pipeline, sr *search.Results) {
-	for _, e := range i.searchCompleteBlocks {
+func (i *instance) searchLocalBlocks(ctx context.Context, req *tempopb.SearchRequest, sr *search.Results) {
+	for _, e := range i.completeBlocks {
+		if e.BlockMeta().Version == v2.Encoding {
```

Comment:
Would lean towards removing the encoding-specific logic from ingesters, and instead catching errUnsupported somewhere.  I see that this avoids starting up a bunch of goroutines destined to fail.  Wondering if there is a decent way to push the errUnsupported further up to `tempodb/encoding/versioned.go` instead of the block.XXX methods.

Reply:
we have to decide how we are going to deprecate v2 blocks. if we release 2.0 w/ no support for searching v2 blocks then we can just make Search return err.Unsupported and remove this code.

If we want to release 2.0 with support for backend search on v2 blocks then we need to leave that code and do something like this. I was thinking we would support search on the backend v2 blocks for at least a minor rev which is why I went this direction.

Reply:
Yes agree on the transition period for v2.  And good with the new flag to control flatbuffer search and startup warnings instead of pushing the logic to the encodings.